Error Creating Alignment Tree for 2016 Data

Hopefully this is an issue with how I've configured my run, but we shall see. I am running with hps-java:master (commit 7e9a7ea) and my branch of hps-mc developing an alignment workflow ([tomeichlersmith:342-alignment-workflow-components](https://github.com/tomeichlersmith/hps-mc/tree/342-alignment-workflow-components) commit [8e00404](https://github.com/tomeichlersmith/hps-mc/commit/8e004042b1260d1f6bbd887becbacd02048b0a4c)).

@cbravo135 @pbutti Do I need to provide some sort of detector map that was previously being provided automatically for the 2019 data I was test-running on? Or do I need to update the hps-java code to be able to handle the 2016 detector's alignment tree?

I am attempting to run the KF-based tracking and alignment over a subsample of 2016 data but I get the following error when attempting to run over a single partition.
```
java.lang.NullPointerException: Cannot invoke "org.hps.recon.tracking.gbl.AlignableVolume.getL2G()" because "ref_vol" is null
        at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.buildComposite(AlignmentStructuresBuilder.java:415)
        at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.<init>(AlignmentStructuresBuilder.java:114)
        at org.hps.recon.tracking.gbl.SimpleGBLTrajAliDriver.detectorChanged(SimpleGBLTrajAliDriver.java:455)
        at org.lcsim.util.Driver.detectorChanged(Driver.java:170)
        at org.lcsim.util.DriverAdapter.detectorChanged(DriverAdapter.java:86)
        at org.lcsim.util.DriverAdapter.recordSupplied(DriverAdapter.java:72)
        at org.freehep.record.loop.DefaultRecordLoop.consumeRecord(DefaultRecordLoop.java:832)
        at org.freehep.record.loop.DefaultRecordLoop.loop(DefaultRecordLoop.java:668)
        at org.freehep.record.loop.DefaultRecordLoop.execute(DefaultRecordLoop.java:566)
        at org.lcsim.util.loop.LCSimLoop.loop(LCSimLoop.java:152)
        at org.lcsim.job.JobControlManager.run(JobControlManager.java:949)
        at org.hps.job.JobManager.main(JobManager.java:31)
```
I've tracked this down for us to look at.

https://github.com/JeffersonLab/hps-java/blob/7e9a7eac08b6229d4f8b8bf02e29cbcc432215b4/tracking/src/main/java/org/hps/recon/tracking/gbl/AlignmentStructuresBuilder.java#L407-L419


<details>
  <summary>Full Run Log</summary>

```
hpsmc:INFO Config files found: ['/sdf/home/e/eichl008/.hpsmc']
hpsmc.job:INFO Job ID: 1
hpsmc.job:INFO Description: HPS MC Job
hpsmc.job:INFO Job configuration:
[MG4]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
madgraph_dir=/sdf/group/hps/users/eichl008/hps/mc/generators/madgraph4/src

[MG5]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
madgraph_dir=/sdf/group/hps/users/eichl008/hps/mc/generators/madgraph5/src

[EGS5]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
egs5_dir=/sdf/group/hps/users/eichl008/hps/mc/generators/egs5

[StdHepConverter]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
egs5_dir=/sdf/group/hps/users/eichl008/hps/mc/generators/egs5

[SLIC]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
slic_dir=/sdf/group/hps/users/bravo/src/slic/install
hps_fieldmaps_dir=/sdf/group/hps/users/bravo/src/hps-fieldmaps
detector_dir=/sdf/group/hps/users/eichl008/hps/java/detector-data/detectors

[JobManager]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
hps_fieldmaps_dir=/sdf/group/hps/users/bravo/src/hps-fieldmaps
java_args=-Xmx3g -XX:+UseSerialGC -Djna.library.path=/sdf/group/hps/users/pbutti/sw/GeneralBrokenLines/cpp/install/lib/

[HPSTR]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
hpstr_install_dir=/sdf/group/hps/users/bravo/src/hpstr/install
hpstr_base=/sdf/group/hps/users/bravo/src/hpstr

[LCIOCount]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
lcio_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/lcsim/lcio/2.7.4-SNAPSHOT/lcio-2.7.4-SNAPSHOT-bin.jar

[LCIOMerge]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
lcio_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/lcsim/lcio/2.7.4-SNAPSHOT/lcio-2.7.4-SNAPSHOT-bin.jar

[EvioToLcio]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
java_args=-Xmx3g -XX:+UseSerialGC

[PEDE]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
param_map=/sdf/group/hps/users/eichl008/hps/alignment/hpsSvtParamMap_2019.txt
pede_minimization=/sdf/group/hps/users/eichl008/hps/alignment/pede_steer_minimization_template.txt

[ApplyPedeRes]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
java_dir=/sdf/group/hps/users/eichl008/hps/java

[ConstructDetector]
hps_home=/sdf/group/hps/users/eichl008/hps
hps_java_bin_jar=/sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar
java_dir=/sdf/group/hps/users/eichl008/hps/java

hpsmc.job:INFO Loading job script: /sdf/group/hps/users/eichl008/hps/mc/install/lib/python/jobs/track_align_job.py
hpsmc.job:INFO Job components loaded: ['job_manager']
hpsmc.job:INFO Job parameters loaded: {'detector': 'HPS-PhysicsRun2016-Pass2', 'run_number': 7800, 'steering': 'kf', 'input_files': {'/sdf/group/hps/data/physrun2016/recon/HPS-PhysicsRun2016-Pass2/hps_007800/hps_007800.evio.322.slcio': 'events.slcio'}, 'output_files': {'events_kf_mille.bin': 'hps_007800.evio.322_mille.bin', 'events_kf_mille_gblplots.root': 'hps_007800.evio.322_gblplots.root', 'mille_constraint.txt': 'hps_007800.evio.322_mille_constraint.txt'}, 'output_dir': '/sdf/group/hps/users/eichl008/hps/2016-align/output/kf/HPS-PhysicsRun2016-Pass2', 'steering_files': {'kf': '/sdf/group/hps/users/eichl008/hps/mc/examples/alignment/tracking/tracking_kf_alignment.lcsim'}, 'job_id': 1}
hpsmc.job:WARNING Config section not found: Job
hpsmc.job:INFO Changing to run dir: /sdf/group/hps/users/eichl008/hps/2016-align/tracking
hpsmc.job:INFO Copying input file: /sdf/group/hps/data/physrun2016/recon/HPS-PhysicsRun2016-Pass2/hps_007800/hps_007800.evio.322.slcio -> /sdf/group/hps/users/eichl008/hps/2016-align/tracking/events.slcio
hpsmc.job:INFO The input file 'events.slcio' already exists at destination '/sdf/group/hps/users/eichl008/hps/2016-align/tracking'
hpsmc.job:INFO Executing 'job_manager' with command: java -Xmx3g -XX:+UseSerialGC -Djna.library.path=/sdf/group/hps/users/pbutti/sw/GeneralBrokenLines/cpp/install/lib/ -jar /sdf/home/e/eichl008/.m2/repository/org/hps/hps-distribution/5.2-SNAPSHOT/hps-distribution-5.2-SNAPSHOT-bin.jar -R 7800 -d HPS-PhysicsRun2016-Pass2 -D outputFile=events_kf_mille -D enableAlignmentCuts=True -D doCOMAlignment=True -D minMom=0.5 -D maxMom=5.0 -D nHitsCut=6 -D debugAlignmentDs=False -D correctTrack=True -D includeNoHitScatters=False -D gblRefitIterations=0 -D storeTrackStates=True -D compositeAlign=True -D constrainedFit=False -D momC=4.55 -D constrainedBSFit=False -D bsZ=-7.7 -D trackSide=-1 -D writeMilleChisqCut=5 -D enableStandardCuts=False -D maxTrackChisqFourHits=60.0 -D maxTrackChisqFiveHits=60.0 -D maxTrackChisqSixHits=60.0 -D inputCollectionName=KalmanFullTracks /sdf/group/hps/users/eichl008/hps/mc/examples/alignment/tracking/tracking_kf_alignment.lcsim -i events.slcio
================ Component: job_manager ================
Reading default logging config from logging.properties
log4j:WARN No appenders could be found for logger (org.reflections.Reflections).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
org.lcsim.job:CONFIG parsing command line arguments
org.lcsim.job:CONFIG outputFile = events_kf_mille
org.lcsim.job:CONFIG defined outputFile = events_kf_mille
org.lcsim.job:CONFIG enableAlignmentCuts = True
org.lcsim.job:CONFIG defined enableAlignmentCuts = True
org.lcsim.job:CONFIG doCOMAlignment = True
org.lcsim.job:CONFIG defined doCOMAlignment = True
org.lcsim.job:CONFIG minMom = 0.5
org.lcsim.job:CONFIG defined minMom = 0.5
org.lcsim.job:CONFIG maxMom = 5.0
org.lcsim.job:CONFIG defined maxMom = 5.0
org.lcsim.job:CONFIG nHitsCut = 6
org.lcsim.job:CONFIG defined nHitsCut = 6
org.lcsim.job:CONFIG debugAlignmentDs = False
org.lcsim.job:CONFIG defined debugAlignmentDs = False
org.lcsim.job:CONFIG correctTrack = True
org.lcsim.job:CONFIG defined correctTrack = True
org.lcsim.job:CONFIG includeNoHitScatters = False
org.lcsim.job:CONFIG defined includeNoHitScatters = False
org.lcsim.job:CONFIG gblRefitIterations = 0
org.lcsim.job:CONFIG defined gblRefitIterations = 0
org.lcsim.job:CONFIG storeTrackStates = True
org.lcsim.job:CONFIG defined storeTrackStates = True
org.lcsim.job:CONFIG compositeAlign = True
org.lcsim.job:CONFIG defined compositeAlign = True
org.lcsim.job:CONFIG constrainedFit = False
org.lcsim.job:CONFIG defined constrainedFit = False
org.lcsim.job:CONFIG momC = 4.55
org.lcsim.job:CONFIG defined momC = 4.55
org.lcsim.job:CONFIG constrainedBSFit = False
org.lcsim.job:CONFIG defined constrainedBSFit = False
org.lcsim.job:CONFIG bsZ = -7.7
org.lcsim.job:CONFIG defined bsZ = -7.7
org.lcsim.job:CONFIG trackSide = -1
org.lcsim.job:CONFIG defined trackSide = -1
org.lcsim.job:CONFIG writeMilleChisqCut = 5
org.lcsim.job:CONFIG defined writeMilleChisqCut = 5
org.lcsim.job:CONFIG enableStandardCuts = False
org.lcsim.job:CONFIG defined enableStandardCuts = False
org.lcsim.job:CONFIG maxTrackChisqFourHits = 60.0
org.lcsim.job:CONFIG defined maxTrackChisqFourHits = 60.0
org.lcsim.job:CONFIG maxTrackChisqFiveHits = 60.0
org.lcsim.job:CONFIG defined maxTrackChisqFiveHits = 60.0
org.lcsim.job:CONFIG maxTrackChisqSixHits = 60.0
org.lcsim.job:CONFIG defined maxTrackChisqSixHits = 60.0
org.lcsim.job:CONFIG inputCollectionName = KalmanFullTracks
org.lcsim.job:CONFIG defined inputCollectionName = KalmanFullTracks
org.lcsim.job:CONFIG added input file events.slcio
org.lcsim.job:CONFIG detector: HPS-PhysicsRun2016-Pass2
org.lcsim.job:CONFIG runNumber: 7800
org.lcsim.job:CONFIG initializing from steering file /sdf/group/hps/users/eichl008/hps/mc/examples/alignment/tracking/tracking_kf_alignment.lcsim
org.lcsim.job:CONFIG initializing LCSim loop
org.lcsim.job:CONFIG EventPrintLoopAdapter is disabled.
SimpleGBLTrajAliDriver::WARNING:Enabling standardCuts!
org.lcsim.job:CONFIG No input files in XML file.
org.lcsim.job:INFO running job
org.lcsim.job:INFO Job started: Fri Mar 31 10:52:50 PDT 2023
org.hps.conditions.database:CONFIG Initializing conditions system with detector 'HPS-PhysicsRun2016-Pass2' and run 7800
org.hps.conditions.database:INFO Opening conditions db connection ...
url: jdbc:mysql://hpsdb.jlab.org:3306/hps_conditions
user: hpsuser
org.lcsim.geometry.compact.converter:INFO HPSTracker2014v1GeometryDefinition: Constructed 14 geometry objects
org.lcsim.geometry.compact.converter:INFO HPSTracker2014v1GeometryDefinition: Constructed 12 module bundles
org.lcsim.detector.converter.compact:INFO creating detector element for subdetector Ecal
org.lcsim.detector.converter.compact:INFO converting subdetector Ecal
org.hps.detector.svt:INFO activating default setup
org.hps.conditions.database:INFO loading conditions set...
id: 314
name: svt_channels
runStart: 2000
runEnd: 999999
tableName: svt_channels
collectionId: 2
updated: 2015-05-04 20:26:37.0
created: 2015-03-15 23:44:16.0
tag: eng_run
createdBy: omoreno
notes: Engineering run SVT channel IDs. Loaded using SvtConditionsLoader.

org.hps.conditions.database:INFO loading conditions set...
id: 313
name: svt_daq_map
runStart: 2000
runEnd: 999999
tableName: svt_daq_map
collectionId: 2
updated: 2015-05-04 20:26:37.0
created: 2015-03-15 23:44:05.0
tag: eng_run
createdBy: omoreno
notes: Engineering run DAQ map. Loaded using SvtConditionsLoader.

org.hps.conditions.database:INFO loading conditions set...
id: 1316
name: svt_t0_shifts
runStart: 2000
runEnd: 9999
tableName: svt_t0_shifts
collectionId: 1940
updated: 2015-12-03 00:45:02.0
created: 2015-12-02 19:45:02.0
tag: null
createdBy: uemura
notes: null

org.hps.conditions.database:INFO loading conditions set...
id: 1397
name: svt_calibrations
runStart: 7566
runEnd: 8063
tableName: svt_calibrations
collectionId: 20
updated: 2017-05-31 18:06:36.0
created: 2016-02-26 14:42:17.0
tag: eng_run
createdBy: phansson
notes: Pedestals and noise. Loaded using SvtConditionsLoader.

org.hps.conditions.database:INFO loading conditions set...
id: 1342
name: svt_shape_fit_parameters
runStart: 7486
runEnd: 9000
tableName: svt_shape_fit_parameters
collectionId: 1959
updated: 2016-02-22 01:56:13.0
created: 2016-02-12 21:02:27.0
tag: null
createdBy: meeg
notes: null

org.hps.conditions.database:INFO loading conditions set...
id: 315
name: svt_gains
runStart: 2000
runEnd: 999999
tableName: svt_gains
collectionId: 2
updated: 2015-05-04 20:26:12.0
created: 2015-03-26 00:26:33.0
tag: eng_run
createdBy: omoreno
notes: Dummy engineering run SVT channel gains.

org.hps.detector.svt:INFO loading default SVT conditions onto subdetector Tracker
org.hps.detector.svt:INFO setting up 36 SVT sensors
org.hps.detector.svt:INFO channel map has 23040 entries
org.hps.conditions.database:INFO Closing database connection.
org.hps.conditions.database:CONFIG Conditions system is now frozen and will not accept updates to detector or run.
org.hps.recon.ecal:CONFIG minLookbackEvents: 5
org.hps.recon.ecal:CONFIG maxLookbackEvents: 40
org.hps.recon.ecal:CONFIG maxLookbackTime:-1
org.hps.conditions.database:INFO Opening conditions db connection ...
url: jdbc:mysql://hpsdb.jlab.org:3306/hps_conditions
user: hpsuser
org.hps.conditions.database:INFO loading conditions set...
id: 1423
name: ecal_channels
runStart: 7032
runEnd: 7874
tableName: ecal_channels
collectionId: 2017
updated: 2016-03-17 21:20:44.0
created: 2016-03-17 17:20:43.0
tag: null
createdBy: baltzell
notes: swapping

org.hps.conditions.database:INFO loading conditions set...
id: 2388
name: ecal_gains
runStart: 7000
runEnd: 9999
tableName: ecal_gains
collectionId: 2976
updated: 2016-11-01 18:43:58.0
created: 2016-11-01 14:43:58.0
tag: null
createdBy: holly
notes: adjusted single channel gain which was very low for testing

org.hps.conditions.database:INFO loading conditions set...
id: 1354
name: ecal_calibrations
runStart: 7350
runEnd: 99999
tableName: ecal_calibrations
collectionId: 1970
updated: 2016-02-21 16:03:28.0
created: 2016-02-21 11:03:27.0
tag: null
createdBy: baltzell
notes: 200nA peds, DAC3280, nosplitter

org.hps.conditions.database:INFO loading conditions set...
id: 1424
name: ecal_time_shifts
runStart: 7000
runEnd: 99999
tableName: ecal_time_shifts
collectionId: 2018
updated: 2016-03-21 14:47:10.0
created: 2016-03-21 10:47:09.0
tag: null
createdBy: holly
notes: time offsets corrected for cable swap

org.hps.conditions.database:INFO loading conditions set...
id: 414
name: ecal_pulse_widths
runStart: 0
runEnd: 99999999
tableName: ecal_pulse_widths
collectionId: 1034
updated: 2015-07-07 21:53:01.0
created: 2015-07-07 17:53:01.0
tag: null
createdBy: nab
notes: 3-pole pulse widths measured from 2015 data

org.hps.conditions.database:INFO loading conditions set...
id: 1422
name: ecal_time_walk
runStart: 7000
runEnd: 99999
tableName: ecal_time_walk
collectionId: 2016
updated: 2016-03-17 13:49:52.0
created: 2016-03-17 09:49:51.0
tag: null
createdBy: holly
notes: time walk parameters for 2016 run

KalmanPatRecDriver: entering detectorChanged, logger level = WARNING

KalmanParams: dump of the Kalman pattern recognition cuts and parameters
  (In the case of two values, they refer to the two iterations.)
  There are 14 layers in the tracker.
  Cluster energy cuts for seeds, by layer:    1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30
  Residual improvement ratio necessary to use a low-ph hit instead of high-ph =     0.25
  First layer in the tracking system: 0
  Number of global iterations: 2
  Number of Kalman filter iterations per track in the final fits: 1
  Detector edge tolerance =    1.000
  Maximum seed curvature=    4.00,     8.00 (1/GeV)
  Minimum seed curvature=    0.00 (1/GeV)
  Maximum tan(lambda):     0.10,     0.13
  Maximum dRho at target plane for seeds:    15.00,    25.00 (mm)
  Maximum dz at target plane for seeds:     3.00,     7.50 (mm)
  Maximum chi^2/hit for a good track:    16.00,    32.00
  Minimum number of hits in the initial outward filtering: 5
  Minimum number of hits for a good track: 7, 6
  Minimum number of stereo hits: 4 3
  Minimum number of axial hits: 2
  Maximum chi^2 increment to add a hit to a track, or minimum to remove a hit:    10.00
  Chi^2 increment threshold for removing a bad hit from a track candidate:    10.00
  Maximum residual, in units of detector resolution, for picking up a hit:    50.00,   100.00
  Maximum residual, in units of detector resolution, for a hit to be shared:    20.00
  Maximum chi^2 increment to keep a shared hit:     6.00
  Maximum number of shared hits on a track: 2
  Maximum time difference among the hits on a track:    30.00 ns
  Threshold to remove redundant seeds (-1 to disable):     0.05
  Maximum chi^2 for 5-hit tracks with a vertex constraint:     1.00
  Default origin to use for vertex constraints:
      0:    0.000 +-    0.050
      1:    0.000 +-    1.000
      2:    0.000 +-    0.020
  Search strategies for bottom:
      6  7  8  9  10 
      4  5  6  7  8 
      5  6  8  9  10 
      5  6  7  8  10 
      3  6  8  9  10 
      4  5  8  9  10 
      4  6  7  8  11 
      4  6  7  8  9 
      4  6  7  9  10 
      2  5  8  9  10 
      8  10  11  12  13 
      6  9  10  11  12 
      2  3  4  5  6 
      2  4  5  6  7 
      8  9  10  11  12 
      6  8  10  11  13 
      6  7  8  10  13 
      1  2  3  4  6 
      0  2  3  4  5 
      0  3  4  5  6 
  Search strategies for top:
      6  7  8  9  11 
      4  5  6  7  9 
      4  7  8  9  11 
      4  6  7  9  11 
      2  7  8  9  11 
      4  5  8  9  11 
      5  6  7  9  10 
      5  6  7  8  9 
      5  6  7  8  11 
      3  4  8  9  11 
      9  10  11  12  13 
      7  8  10  11  13 
      2  3  4  5  7 
      3  4  5  6  7 
      8  9  10  11  13 
      7  9  10  11  12 
      6  7  9  11  12 
      0  2  3  5  7 
      1  2  3  4  5 
      1  2  4  5  7 
  The number of seed strategies used in the first iteration is 17

org.hps.recon.tracking.kalman.KalmanInterface:INFO Entering the KalmanInterface constructor
org.hps.recon.tracking.kalman.KalmanPatRecHPS:INFO KalmanPatRecHPS instance created.
org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 2 Det= 1 Top=false Mpd=1 split=false stereo=true pnt=   4.479  103.649   20.690 t= -0.031  -1.000   0.001
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 2 Det= 0 Top=true Mpd=1 split=false stereo=false pnt=   3.408   87.920  -20.710 t= -0.031  -1.000   0.001
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 3 Det= 1 Top=false Mpd=2 split=false stereo=false pnt=   4.774  111.710   20.742 t=  0.029   1.000   0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 3 Det= 0 Top=true Mpd=2 split=false stereo=true pnt=   3.695   96.105  -20.677 t=  0.030   1.000  -0.001
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 4 Det= 1 Top=false Mpd=3 split=false stereo=true pnt=   7.657  203.771   22.178 t= -0.030  -1.000   0.006
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 4 Det= 0 Top=true Mpd=3 split=false stereo=false pnt=   6.434  187.993  -22.245 t= -0.031  -1.000  -0.005
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 5 Det= 1 Top=false Mpd=4 split=false stereo=false pnt=   7.765  211.826   22.214 t=  0.029   1.000   0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 5 Det= 0 Top=true Mpd=4 split=false stereo=true pnt=   6.654  195.990  -22.205 t=  0.030   1.000  -0.008
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 6 Det= 1 Top=false Mpd=5 split=false stereo=true pnt=  10.483  303.721   23.670 t= -0.029  -1.000   0.005
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 6 Det= 0 Top=true Mpd=5 split=false stereo=false pnt=   9.398  287.978  -23.753 t= -0.031  -1.000   0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 7 Det= 1 Top=false Mpd=6 split=false stereo=false pnt=  10.711  311.588   23.684 t=  0.028   1.000  -0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 7 Det= 0 Top=true Mpd=6 split=false stereo=true pnt=   9.635  295.966  -23.714 t=  0.029   1.000  -0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 1 Top=false Mpd=7 split=false stereo=true pnt= -34.085  505.574   29.193 t= -0.029  -1.000   0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 3 Top=false Mpd=9 split=false stereo=true pnt=  66.793  502.523   24.163 t= -0.030  -1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 0 Top=true Mpd=7 split=false stereo=false pnt= -35.463  489.651  -26.645 t= -0.031  -1.000   0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 2 Top=true Mpd=9 split=false stereo=false pnt=  65.414  486.509  -26.647 t= -0.031  -1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 1 Top=false Mpd=8 split=false stereo=false pnt= -33.848  513.077   26.691 t=  0.030   1.000  -0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 3 Top=false Mpd=10 split=false stereo=false pnt=  66.958  510.026   26.686 t=  0.030   1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 0 Top=true Mpd=8 split=false stereo=true pnt= -35.171  497.116  -29.168 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 2 Top=true Mpd=10 split=false stereo=true pnt=  65.578  493.992  -24.156 t=  0.030   1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 1 Top=false Mpd=11 split=false stereo=true pnt= -28.009  705.467   32.209 t= -0.031  -1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 3 Top=false Mpd=13 split=false stereo=true pnt=  72.778  702.428   27.179 t= -0.030  -1.000  -0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 0 Top=true Mpd=11 split=false stereo=false pnt= -29.417  689.484  -29.722 t= -0.032  -0.999   0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 2 Top=true Mpd=13 split=false stereo=false pnt=  71.419  686.370  -29.741 t= -0.029  -1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 1 Top=false Mpd=12 split=false stereo=false pnt= -27.846  713.066   29.684 t=  0.030   1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 3 Top=false Mpd=14 split=false stereo=false pnt=  73.034  710.031   29.687 t=  0.031   1.000   0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 0 Top=true Mpd=12 split=false stereo=true pnt= -29.064  697.092  -32.219 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 2 Top=true Mpd=14 split=false stereo=true pnt=  71.703  693.940  -27.215 t=  0.031   1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 1 Top=false Mpd=15 split=false stereo=true pnt= -21.879  905.349   35.144 t= -0.029  -1.000   0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 3 Top=false Mpd=17 split=false stereo=true pnt=  78.870  902.353   30.136 t= -0.030  -1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 0 Top=true Mpd=15 split=false stereo=false pnt= -23.248  889.458  -32.730 t= -0.030  -1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 2 Top=true Mpd=17 split=false stereo=false pnt=  77.486  886.454  -32.715 t= -0.029  -1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 1 Top=false Mpd=16 split=false stereo=false pnt= -21.795  912.893   32.653 t=  0.028   1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 3 Top=false Mpd=18 split=false stereo=false pnt=  79.097  909.993   32.660 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 0 Top=true Mpd=16 split=false stereo=true pnt= -22.898  896.913  -35.202 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 2 Top=true Mpd=18 split=false stereo=true pnt=  77.728  893.879  -30.215 t=  0.028   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700


KalmanParams: dump of the Kalman pattern recognition cuts and parameters
  (In the case of two values, they refer to the two iterations.)
  There are 14 layers in the tracker.
  Cluster energy cuts for seeds, by layer:    1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30   1.30
  Residual improvement ratio necessary to use a low-ph hit instead of high-ph =     0.25
  First layer in the tracking system: 0
  Number of global iterations: 2
  Number of Kalman filter iterations per track in the final fits: 1
  Detector edge tolerance =    1.000
  Maximum seed curvature=    4.00,     8.00 (1/GeV)
  Minimum seed curvature=    0.00 (1/GeV)
  Maximum tan(lambda):     0.10,     0.13
  Maximum dRho at target plane for seeds:    15.00,    25.00 (mm)
  Maximum dz at target plane for seeds:     3.00,     7.50 (mm)
  Maximum chi^2/hit for a good track:    16.00,    32.00
  Minimum number of hits in the initial outward filtering: 5
  Minimum number of hits for a good track: 7, 6
  Minimum number of stereo hits: 4 3
  Minimum number of axial hits: 2
  Maximum chi^2 increment to add a hit to a track, or minimum to remove a hit:    10.00
  Chi^2 increment threshold for removing a bad hit from a track candidate:    10.00
  Maximum residual, in units of detector resolution, for picking up a hit:    50.00,   100.00
  Maximum residual, in units of detector resolution, for a hit to be shared:    20.00
  Maximum chi^2 increment to keep a shared hit:     6.00
  Maximum number of shared hits on a track: 2
  Maximum time difference among the hits on a track:    30.00 ns
  Threshold to remove redundant seeds (-1 to disable):     0.05
  Maximum chi^2 for 5-hit tracks with a vertex constraint:     1.00
  Default origin to use for vertex constraints:
      0:    0.000 +-    0.050
      1:    0.000 +-    1.000
      2:    0.000 +-    0.020
  Search strategies for bottom:
      6  7  8  9  10 
      4  5  6  7  8 
      5  6  8  9  10 
      5  6  7  8  10 
      3  6  8  9  10 
      4  5  8  9  10 
      4  6  7  8  11 
      4  6  7  8  9 
      4  6  7  9  10 
      2  5  8  9  10 
      8  10  11  12  13 
      6  9  10  11  12 
      2  3  4  5  6 
      2  4  5  6  7 
      8  9  10  11  12 
      6  8  10  11  13 
      6  7  8  10  13 
      1  2  3  4  6 
      0  2  3  4  5 
      0  3  4  5  6 
  Search strategies for top:
      6  7  8  9  11 
      4  5  6  7  9 
      4  7  8  9  11 
      4  6  7  9  11 
      2  7  8  9  11 
      4  5  8  9  11 
      5  6  7  9  10 
      5  6  7  8  9 
      5  6  7  8  11 
      3  4  8  9  11 
      9  10  11  12  13 
      7  8  10  11  13 
      2  3  4  5  7 
      3  4  5  6  7 
      8  9  10  11  13 
      7  9  10  11  12 
      6  7  9  11  12 
      0  2  3  5  7 
      1  2  3  4  5 
      1  2  4  5  7 
  The number of seed strategies used in the first iteration is 17

org.hps.recon.tracking.kalman.KalmanInterface:INFO Entering the KalmanInterface constructor
org.hps.recon.tracking.kalman.KalmanPatRecHPS:INFO KalmanPatRecHPS instance created.
org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 2 Det= 1 Top=false Mpd=1 split=false stereo=true pnt=   4.479  103.649   20.690 t= -0.031  -1.000   0.001
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 2 Det= 0 Top=true Mpd=1 split=false stereo=false pnt=   3.408   87.920  -20.710 t= -0.031  -1.000   0.001
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 3 Det= 1 Top=false Mpd=2 split=false stereo=false pnt=   4.774  111.710   20.742 t=  0.029   1.000   0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 3 Det= 0 Top=true Mpd=2 split=false stereo=true pnt=   3.695   96.105  -20.677 t=  0.030   1.000  -0.001
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 4 Det= 1 Top=false Mpd=3 split=false stereo=true pnt=   7.657  203.771   22.178 t= -0.030  -1.000   0.006
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 4 Det= 0 Top=true Mpd=3 split=false stereo=false pnt=   6.434  187.993  -22.245 t= -0.031  -1.000  -0.005
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 5 Det= 1 Top=false Mpd=4 split=false stereo=false pnt=   7.765  211.826   22.214 t=  0.029   1.000   0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 5 Det= 0 Top=true Mpd=4 split=false stereo=true pnt=   6.654  195.990  -22.205 t=  0.030   1.000  -0.008
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 6 Det= 1 Top=false Mpd=5 split=false stereo=true pnt=  10.483  303.721   23.670 t= -0.029  -1.000   0.005
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 6 Det= 0 Top=true Mpd=5 split=false stereo=false pnt=   9.398  287.978  -23.753 t= -0.031  -1.000   0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 7 Det= 1 Top=false Mpd=6 split=false stereo=false pnt=  10.711  311.588   23.684 t=  0.028   1.000  -0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 7 Det= 0 Top=true Mpd=6 split=false stereo=true pnt=   9.635  295.966  -23.714 t=  0.029   1.000  -0.003
           thick=  0.3200 mm, x ext=-100.0000 100.0000, y ext=-23.5850  23.5850

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 1 Top=false Mpd=7 split=false stereo=true pnt= -34.085  505.574   29.193 t= -0.029  -1.000   0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 3 Top=false Mpd=9 split=false stereo=true pnt=  66.793  502.523   24.163 t= -0.030  -1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 0 Top=true Mpd=7 split=false stereo=false pnt= -35.463  489.651  -26.645 t= -0.031  -1.000   0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 8 Det= 2 Top=true Mpd=9 split=false stereo=false pnt=  65.414  486.509  -26.647 t= -0.031  -1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 1 Top=false Mpd=8 split=false stereo=false pnt= -33.848  513.077   26.691 t=  0.030   1.000  -0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 3 Top=false Mpd=10 split=false stereo=false pnt=  66.958  510.026   26.686 t=  0.030   1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 0 Top=true Mpd=8 split=false stereo=true pnt= -35.171  497.116  -29.168 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr= 9 Det= 2 Top=true Mpd=10 split=false stereo=true pnt=  65.578  493.992  -24.156 t=  0.030   1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 1 Top=false Mpd=11 split=false stereo=true pnt= -28.009  705.467   32.209 t= -0.031  -1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 3 Top=false Mpd=13 split=false stereo=true pnt=  72.778  702.428   27.179 t= -0.030  -1.000  -0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 0 Top=true Mpd=11 split=false stereo=false pnt= -29.417  689.484  -29.722 t= -0.032  -0.999   0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=10 Det= 2 Top=true Mpd=13 split=false stereo=false pnt=  71.419  686.370  -29.741 t= -0.029  -1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 1 Top=false Mpd=12 split=false stereo=false pnt= -27.846  713.066   29.684 t=  0.030   1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 3 Top=false Mpd=14 split=false stereo=false pnt=  73.034  710.031   29.687 t=  0.031   1.000   0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 0 Top=true Mpd=12 split=false stereo=true pnt= -29.064  697.092  -32.219 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=11 Det= 2 Top=true Mpd=14 split=false stereo=true pnt=  71.703  693.940  -27.215 t=  0.031   1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 1 Top=false Mpd=15 split=false stereo=true pnt= -21.879  905.349   35.144 t= -0.029  -1.000   0.002
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 3 Top=false Mpd=17 split=false stereo=true pnt=  78.870  902.353   30.136 t= -0.030  -1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 0 Top=true Mpd=15 split=false stereo=false pnt= -23.248  889.458  -32.730 t= -0.030  -1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=12 Det= 2 Top=true Mpd=17 split=false stereo=false pnt=  77.486  886.454  -32.715 t= -0.029  -1.000   0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 1 Top=false Mpd=16 split=false stereo=false pnt= -21.795  912.893   32.653 t=  0.028   1.000  -0.001
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 3 Top=false Mpd=18 split=false stereo=false pnt=  79.097  909.993   32.660 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 0 Top=true Mpd=16 split=false stereo=true pnt= -22.898  896.913  -35.202 t=  0.031   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

org.hps.recon.tracking.kalman.KalmanInterface:INFO Si Module: Lyr=13 Det= 2 Top=true Mpd=18 split=false stereo=true pnt=  77.728  893.879  -30.215 t=  0.028   1.000  -0.000
           thick=  0.3200 mm, x ext=-50.0000  50.0000, y ext=-20.1700  20.1700

Alignable Detector Elements informations: Tracker_base_alignable 
The parent is: Tracker
Alignable Detector Elements informations: Tracker_alignable_supportRing80_bottom 
The parent is: Tracker_base_alignable
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer1_bottom 
The parent is: Tracker_alignable_supportRing80_bottom
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer2_bottom 
The parent is: Tracker_alignable_supportRing80_bottom
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer3_bottom 
The parent is: Tracker_alignable_supportRing80_bottom
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer4_bottom 
The parent is: Tracker_alignable_supportRing80_bottom
Alignable Detector Elements informations: Tracker_alignable_UChannel90_bottom 
The parent is: Tracker_base_alignable
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer5_bottom 
The parent is: Tracker_alignable_UChannel90_bottom
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer6_bottom 
The parent is: Tracker_alignable_UChannel90_bottom
Alignable Detector Elements informations: Tracker_alignable_supportRing80_top 
The parent is: Tracker_base_alignable
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer1_top 
The parent is: Tracker_alignable_supportRing80_top
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer2_top 
The parent is: Tracker_alignable_supportRing80_top
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer3_top 
The parent is: Tracker_alignable_supportRing80_top
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer4_top 
The parent is: Tracker_alignable_supportRing80_top
Alignable Detector Elements informations: Tracker_alignable_UChannel90_top 
The parent is: Tracker_base_alignable
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer5_top 
The parent is: Tracker_alignable_UChannel90_top
Alignable Detector Elements informations: Tracker_alignable_fullmoduleLayer6_top 
The parent is: Tracker_alignable_UChannel90_top
java.lang.NullPointerException: Cannot invoke "org.hps.recon.tracking.gbl.AlignableVolume.getL2G()" because "ref_vol" is null
	at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.buildComposite(AlignmentStructuresBuilder.java:415)
	at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.<init>(AlignmentStructuresBuilder.java:114)
	at org.hps.recon.tracking.gbl.SimpleGBLTrajAliDriver.detectorChanged(SimpleGBLTrajAliDriver.java:455)
	at org.lcsim.util.Driver.detectorChanged(Driver.java:170)
	at org.lcsim.util.DriverAdapter.detectorChanged(DriverAdapter.java:86)
	at org.lcsim.util.DriverAdapter.recordSupplied(DriverAdapter.java:72)
	at org.freehep.record.loop.DefaultRecordLoop.consumeRecord(DefaultRecordLoop.java:832)
	at org.freehep.record.loop.DefaultRecordLoop.loop(DefaultRecordLoop.java:668)
	at org.freehep.record.loop.DefaultRecordLoop.execute(DefaultRecordLoop.java:566)
	at org.lcsim.util.loop.LCSimLoop.loop(LCSimLoop.java:152)
	at org.lcsim.job.JobControlManager.run(JobControlManager.java:949)
	at org.hps.job.JobManager.main(JobManager.java:31)
org.lcsim.job:SEVERE A fatal error occurred during the job
java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "org.hps.recon.tracking.gbl.AlignableVolume.getL2G()" because "ref_vol" is null
	at org.lcsim.util.loop.LCSimLoop.handleClientError(LCSimLoop.java:194)
	at org.freehep.record.loop.DefaultRecordLoop.consumeRecord(DefaultRecordLoop.java:834)
	at org.freehep.record.loop.DefaultRecordLoop.loop(DefaultRecordLoop.java:668)
	at org.freehep.record.loop.DefaultRecordLoop.execute(DefaultRecordLoop.java:566)
	at org.lcsim.util.loop.LCSimLoop.loop(LCSimLoop.java:152)
	at org.lcsim.job.JobControlManager.run(JobControlManager.java:949)
	at org.hps.job.JobManager.main(JobManager.java:31)
Caused by: java.lang.NullPointerException: Cannot invoke "org.hps.recon.tracking.gbl.AlignableVolume.getL2G()" because "ref_vol" is null
	at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.buildComposite(AlignmentStructuresBuilder.java:415)
	at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.<init>(AlignmentStructuresBuilder.java:114)
	at org.hps.recon.tracking.gbl.SimpleGBLTrajAliDriver.detectorChanged(SimpleGBLTrajAliDriver.java:455)
	at org.lcsim.util.Driver.detectorChanged(Driver.java:170)
	at org.lcsim.util.DriverAdapter.detectorChanged(DriverAdapter.java:86)
	at org.lcsim.util.DriverAdapter.recordSupplied(DriverAdapter.java:72)
	at org.freehep.record.loop.DefaultRecordLoop.consumeRecord(DefaultRecordLoop.java:832)
	... 5 more

Exception in thread "main" java.lang.RuntimeException: Fatal error occurred during the job
	at org.lcsim.job.JobControlManager.run(JobControlManager.java:972)
	at org.hps.job.JobManager.main(JobManager.java:31)
Caused by: java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "org.hps.recon.tracking.gbl.AlignableVolume.getL2G()" because "ref_vol" is null
	at org.lcsim.util.loop.LCSimLoop.handleClientError(LCSimLoop.java:194)
	at org.freehep.record.loop.DefaultRecordLoop.consumeRecord(DefaultRecordLoop.java:834)
	at org.freehep.record.loop.DefaultRecordLoop.loop(DefaultRecordLoop.java:668)
	at org.freehep.record.loop.DefaultRecordLoop.execute(DefaultRecordLoop.java:566)
	at org.lcsim.util.loop.LCSimLoop.loop(LCSimLoop.java:152)
	at org.lcsim.job.JobControlManager.run(JobControlManager.java:949)
	... 1 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.hps.recon.tracking.gbl.AlignableVolume.getL2G()" because "ref_vol" is null
	at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.buildComposite(AlignmentStructuresBuilder.java:415)
	at org.hps.recon.tracking.gbl.AlignmentStructuresBuilder.<init>(AlignmentStructuresBuilder.java:114)
	at org.hps.recon.tracking.gbl.SimpleGBLTrajAliDriver.detectorChanged(SimpleGBLTrajAliDriver.java:455)
	at org.lcsim.util.Driver.detectorChanged(Driver.java:170)
	at org.lcsim.util.DriverAdapter.detectorChanged(DriverAdapter.java:86)
	at org.lcsim.util.DriverAdapter.recordSupplied(DriverAdapter.java:72)
	at org.freehep.record.loop.DefaultRecordLoop.consumeRecord(DefaultRecordLoop.java:832)
	... 5 more
hpsmc.job:INFO Execution of job_manager took 18.7231 second(s) with return code: 1
Traceback (most recent call last):
  File "/sdf/group/hps/users/eichl008/hps/mc/install/lib/python/hpsmc/job.py", line 824, in <module>
    job.run()
  File "/sdf/group/hps/users/eichl008/hps/mc/install/lib/python/hpsmc/job.py", line 508, in run
    self._execute()
  File "/sdf/group/hps/users/eichl008/hps/mc/install/lib/python/hpsmc/job.py", line 566, in _execute
    raise Exception("Output file '%s' is missing after execution." % outputfile)
Exception: Output file 'events_kf_mille_gblplots.root' is missing after execution.

```
</details>

--- a/tracking/src/main/java/org/hps/recon/tracking/gbl/SimpleGBLTrajAliDriver.java
+++ b/tracking/src/main/java/org/hps/recon/tracking/gbl/SimpleGBLTrajAliDriver.java
@@ -570,7 +570,7 @@ public class SimpleGBLTrajAliDriver extends Driver {
                 
                 //Align with tracks with at least 6 hits
                 if ((tanLambda > 0 && track.getTrackerHits().size() < actualHitCut) 
-                        || (tanLambda < 0 && track.getTrackerHits().size() < actualHitCut))  {
+                    || (tanLambda < 0 && track.getTrackerHits().size() < actualHitCut))  {
                     continue;
                 }
                 
@@ -876,165 +876,166 @@ public class SimpleGBLTrajAliDriver extends Driver {
                     
                 }//point loop
                 
-                
-                //Make a gblTrajectory with the points with all the composite derivatives + seed and write the record
-                
-                GblTrajectoryJna trajForMPII = null;
-                GblTrajectoryJna trajForMPII_unconstrained = new GblTrajectoryJna(points_on_traj,1,1,1);
-                
-                //seed matrix q/p, yT', xT', xT, yT 
-                SymMatrix seedPrecision = new SymMatrix(5);
-                
-                if (constrainedFit) 
-                    seedPrecision.set(0,0,seed_precision);
-                if (constrainedTanLFit)
-                    seedPrecision.set(1,1,seed_precision);
-                if (constrainedPhi0Fit)
-                    seedPrecision.set(2,2,seed_precision);
-
-                //z0 constraint 
-                //seedPrecision.set(4,4,seed_precision);
-                
-                //d0 constraint
-                //seedPrecision.set(3,3,1000000);
-                
-                if (!constrainedFit && !constrainedTanLFit && !constrainedPhi0Fit && !constrainedD0Fit && !constrainedZ0Fit) {
-                    trajForMPII =  new GblTrajectoryJna(points_on_traj,1,1,1);
-                } else {
-                    trajForMPII = new GblTrajectoryJna(points_on_traj,1,seedPrecision,1,1,1);
-                }
-                
-                if (debugAlignmentDs) trajForMPII.printData();
-                
-                //Fit the trajectory to get the Chi2
-                trajForMPII_unconstrained.fit(Chi2,Ndf, lostWeight,"");
-
-                //Avoid to use tracks with terrible Chi2
-                if (Chi2.getValue() / Ndf.getValue() > writeMilleChi2Cut)
-                    continue;
-                
-                if (writeMilleBinary) {
-                    /*
-                    System.out.println("DEBUG::Tom::Writing track with "
-                        + points_on_traj.size() + " hits to mille binary.");
-                     */
-                    trajForMPII.milleOut(mille);
-                }
-                
-                if (correctTrack) {                
-
-                    //Form the FittedGblTrajectory for the unconstrained fit
-                    FittedGblTrajectory fitTraj = new FittedGblTrajectory(trajForMPII_unconstrained, Chi2.getValue(), Ndf.getValue(), lostWeight.getValue());
-                    
-                    fitTraj.setSensorMap(sensorMap);
-                    fitTraj.setPathLengthMap(pathLengthMap);
-                    
-                    Collection<TrackerHit> hth = track.getTrackerHits();
-                    List<TrackerHit> allHthList = TrackUtils.sortHits(hth);
-                    Pair<Track, GBLKinkData>  newTrack = MakeGblTracks.makeCorrectedTrack(fitTraj, TrackUtils.getHTF(track), allHthList, 0, bfield);
-                    Track gblTrk = newTrack.getFirst();
-
-                    //System.out.println("DEBUG::Tom::Correct GBL track has "+gblTrk.getTrackerHits().size()+" hits");
-                    
-                    if(computeGBLResiduals) {
-                        
-                        List<Double> b_residuals = new ArrayList<Double>();
-                        List<Float>   b_sigmas    = new ArrayList<Float>();
-                        List<Integer> r_sensors   = new ArrayList<Integer>();
-                        
-                        int numData[] = new int[1];
-                        Integer[] sensorsFromMapArray = fitTraj.getSensorMap().keySet().toArray(new Integer[0]);
-                        for (int i_s = 0; i_s < sensorsFromMapArray.length; i_s++) {
-                            int ilabel = sensorsFromMapArray[i_s];
-                            int mpid = fitTraj.getSensorMap().get(ilabel);
-                            List<Double> aResiduals   = new ArrayList<Double>();
-                            List<Double> aMeasErrors  = new ArrayList<Double>();
-                            List<Double> aResErrors   = new ArrayList<Double>();
-                            List<Double> aDownWeights = new ArrayList<Double>();
-                            trajForMPII_unconstrained.getMeasResults(ilabel,numData,aResiduals,aMeasErrors,aResErrors,aDownWeights);
-                            if (numData[0]>1) { 
-                                System.out.printf("GBLRefitterDriver::WARNING::We have SCT sensors. Residuals dimensions should be <=1\n");
-                            }
-                            for (int i=0; i<numData[0];i++) {
-                                //System.out.printf("Example1::ilabel numDataIDX MPID aResidual aMeasError aResError\n");
-                                //System.out.printf("Example1::measResults %d %d %d %f %f %f \n",ilabel, i, mpid, aResiduals.get(i),aMeasErrors.get(i),aResErrors.get(i));
-                                
-                                r_sensors.add(mpid);
-                                b_residuals.add(aResiduals.get(i));
-                                b_sigmas.add(aResErrors.get(i).floatValue());
-                            }
-                            
-                            GblTrajectoryJna gbl_fit_traj_u = new GblTrajectoryJna(points_on_traj,1,1,1);
-                            DoubleByReference Chi2_u = new DoubleByReference(0.);
-                            DoubleByReference lostWeight_u = new DoubleByReference(0.);
-                            IntByReference Ndf_u = new IntByReference(0);
-                            int[] u_numData  = new int[1]; 
-                            //Fit it once to have exactly the same starting point of gbl_fit_trajectory.
-                            gbl_fit_traj_u.fit(Chi2_u,Ndf_u,lostWeight_u,"");
-                            List<Double> u_aResiduals   = new ArrayList<Double>();   
-                            List<Double> u_aMeasErrors  = new ArrayList<Double>();
-                            List<Double> u_aResErrors   = new ArrayList<Double>();  
-                            List<Double> u_aDownWeights = new ArrayList<Double>();
-                            
-                            try {
-                                //Fit removing the measurement
-                                gbl_fit_traj_u.fit(Chi2_u,Ndf_u,lostWeight_u,"",ilabel);
-                                gbl_fit_traj_u.getMeasResults(ilabel,u_numData,u_aResiduals,u_aMeasErrors,u_aResErrors,u_aDownWeights); 
-                                for (int i=0; i<u_numData[0];i++) {
-                                    //System.out.printf("Example1::ilabel numDataIDX MPID aResidual aMeasError aResError\n");
-                                    //System.out.printf("Example1::UmeasResults %d %d %d %f %f %f \n",ilabel, i, mpid, u_aResiduals.get(i),u_aMeasErrors.get(i),u_aResErrors.get(i));
-                                    
-                                    r_sensors.add(mpid);
-                                    b_residuals.add(u_aResiduals.get(i));
-                                    b_sigmas.add(u_aResErrors.get(i).floatValue());
-                                }
-                            }
-                            catch (RuntimeException e){
-                                //  e.printStackTrack();
-                                r_sensors.add(-999);
-                                b_residuals.add(-9999.);
-                                b_sigmas.add((float)-9999.);
-                                //System.out.printf("Unbiasing fit fails! For label::%d\n",ilabel);
-                            }
-                            
-                            
-                            gbl_fit_traj_u.delete();
-                        } // loop on sensor map
-
-
-
-                        int trackerVolume = 0;
-                        if (gblTrk.getTrackStates().get(0).getTanLambda() < 0) trackerVolume = 1;
-                        TrackResidualsData resData  = new TrackResidualsData(trackerVolume,r_sensors,b_residuals,b_sigmas);
-                        trackResidualsCollection.add(resData);
-                        trackResidualsRelations.add(new BaseLCRelation(resData,gblTrk));
-
-                    } //computeResiduals
-                    
-                    
-                    
-                    //System.out.println("Refitted track chi2 " + gblTrk.getChi2());
-                    refittedTracks.add(gblTrk);
-                    kinkDataCollection.add(newTrack.getSecond());
-                    kinkDataRelations.add(new BaseLCRelation(newTrack.getSecond(), gblTrk));
-                }
-                
-                trajForMPII.delete();
-                trajForMPII_unconstrained.delete();
-                
             }// composite Alignment
+                
+            //Make a gblTrajectory with the points with all the composite derivatives + seed and write the record
+                
+            GblTrajectoryJna trajForMPII = null;
+            GblTrajectoryJna trajForMPII_unconstrained = new GblTrajectoryJna(points_on_traj,1,1,1);
+                
+            //seed matrix q/p, yT', xT', xT, yT 
+            SymMatrix seedPrecision = new SymMatrix(5);
+                
+            if (constrainedFit) 
+                seedPrecision.set(0,0,seed_precision);
+            if (constrainedTanLFit)
+                seedPrecision.set(1,1,seed_precision);
+            if (constrainedPhi0Fit)
+                seedPrecision.set(2,2,seed_precision);
+
+            //z0 constraint 
+            //seedPrecision.set(4,4,seed_precision);
+                
+            //d0 constraint
+            //seedPrecision.set(3,3,1000000);
+                
+            if (!constrainedFit && !constrainedTanLFit && !constrainedPhi0Fit && !constrainedD0Fit && !constrainedZ0Fit) {
+                trajForMPII =  new GblTrajectoryJna(points_on_traj,1,1,1);
+            } else {
+                trajForMPII = new GblTrajectoryJna(points_on_traj,1,seedPrecision,1,1,1);
+            }
+                
+            if (debugAlignmentDs) trajForMPII.printData();
+                
+            //Fit the trajectory to get the Chi2
+            trajForMPII_unconstrained.fit(Chi2,Ndf, lostWeight,"");
+
+            //Avoid to use tracks with terrible Chi2
+            if (Chi2.getValue() / Ndf.getValue() > writeMilleChi2Cut)
+                continue;
+                
+            if (writeMilleBinary) {
+                /*
+                  System.out.println("DEBUG::Tom::Writing track with "
+                  + points_on_traj.size() + " hits to mille binary.");
+                */
+                trajForMPII.milleOut(mille);
+            }
+                
+            if (correctTrack) {                
+
+                //Form the FittedGblTrajectory for the unconstrained fit
+                FittedGblTrajectory fitTraj = new FittedGblTrajectory(trajForMPII_unconstrained, Chi2.getValue(), Ndf.getValue(), lostWeight.getValue());
+                    
+                fitTraj.setSensorMap(sensorMap);
+                fitTraj.setPathLengthMap(pathLengthMap);
+                    
+                Collection<TrackerHit> hth = track.getTrackerHits();
+                List<TrackerHit> allHthList = TrackUtils.sortHits(hth);
+                Pair<Track, GBLKinkData>  newTrack = MakeGblTracks.makeCorrectedTrack(fitTraj, TrackUtils.getHTF(track), allHthList, 0, bfield);
+                Track gblTrk = newTrack.getFirst();
+
+                //System.out.println("DEBUG::Tom::Correct GBL track has "+gblTrk.getTrackerHits().size()+" hits");
+                    
+                if(computeGBLResiduals) {
+                        
+                    List<Double> b_residuals = new ArrayList<Double>();
+                    List<Float>   b_sigmas    = new ArrayList<Float>();
+                    List<Integer> r_sensors   = new ArrayList<Integer>();
+                        
+                    int numData[] = new int[1];
+                    Integer[] sensorsFromMapArray = fitTraj.getSensorMap().keySet().toArray(new Integer[0]);
+                    for (int i_s = 0; i_s < sensorsFromMapArray.length; i_s++) {
+                        int ilabel = sensorsFromMapArray[i_s];
+                        int mpid = fitTraj.getSensorMap().get(ilabel);
+                        List<Double> aResiduals   = new ArrayList<Double>();
+                        List<Double> aMeasErrors  = new ArrayList<Double>();
+                        List<Double> aResErrors   = new ArrayList<Double>();
+                        List<Double> aDownWeights = new ArrayList<Double>();
+                        trajForMPII_unconstrained.getMeasResults(ilabel,numData,aResiduals,aMeasErrors,aResErrors,aDownWeights);
+                        if (numData[0]>1) { 
+                            System.out.printf("GBLRefitterDriver::WARNING::We have SCT sensors. Residuals dimensions should be <=1\n");
+                        }
+                        for (int i=0; i<numData[0];i++) {
+                            //System.out.printf("Example1::ilabel numDataIDX MPID aResidual aMeasError aResError\n");
+                            //System.out.printf("Example1::measResults %d %d %d %f %f %f \n",ilabel, i, mpid, aResiduals.get(i),aMeasErrors.get(i),aResErrors.get(i));
+                                
+                            r_sensors.add(mpid);
+                            b_residuals.add(aResiduals.get(i));
+                            b_sigmas.add(aResErrors.get(i).floatValue());
+                        }
+                            
+                        GblTrajectoryJna gbl_fit_traj_u = new GblTrajectoryJna(points_on_traj,1,1,1);
+                        DoubleByReference Chi2_u = new DoubleByReference(0.);
+                        DoubleByReference lostWeight_u = new DoubleByReference(0.);
+                        IntByReference Ndf_u = new IntByReference(0);
+                        int[] u_numData  = new int[1]; 
+                        //Fit it once to have exactly the same starting point of gbl_fit_trajectory.
+                        gbl_fit_traj_u.fit(Chi2_u,Ndf_u,lostWeight_u,"");
+                        List<Double> u_aResiduals   = new ArrayList<Double>();   
+                        List<Double> u_aMeasErrors  = new ArrayList<Double>();
+                        List<Double> u_aResErrors   = new ArrayList<Double>();  
+                        List<Double> u_aDownWeights = new ArrayList<Double>();
+                            
+                        try {
+                            //Fit removing the measurement
+                            gbl_fit_traj_u.fit(Chi2_u,Ndf_u,lostWeight_u,"",ilabel);
+                            gbl_fit_traj_u.getMeasResults(ilabel,u_numData,u_aResiduals,u_aMeasErrors,u_aResErrors,u_aDownWeights); 
+                            for (int i=0; i<u_numData[0];i++) {
+                                //System.out.printf("Example1::ilabel numDataIDX MPID aResidual aMeasError aResError\n");
+                                //System.out.printf("Example1::UmeasResults %d %d %d %f %f %f \n",ilabel, i, mpid, u_aResiduals.get(i),u_aMeasErrors.get(i),u_aResErrors.get(i));
+                                    
+                                r_sensors.add(mpid);
+                                b_residuals.add(u_aResiduals.get(i));
+                                b_sigmas.add(u_aResErrors.get(i).floatValue());
+                            }
+                        }
+                        catch (RuntimeException e){
+                            //  e.printStackTrack();
+                            r_sensors.add(-999);
+                            b_residuals.add(-9999.);
+                            b_sigmas.add((float)-9999.);
+                            //System.out.printf("Unbiasing fit fails! For label::%d\n",ilabel);
+                        }
+                            
+                            
+                        gbl_fit_traj_u.delete();
+                    } // loop on sensor map
+
+
+
+                    int trackerVolume = 0;
+                    if (gblTrk.getTrackStates().get(0).getTanLambda() < 0) trackerVolume = 1;
+                    TrackResidualsData resData  = new TrackResidualsData(trackerVolume,r_sensors,b_residuals,b_sigmas);
+                    trackResidualsCollection.add(resData);
+                    trackResidualsRelations.add(new BaseLCRelation(resData,gblTrk));
+
+                } //computeResiduals
+                    
+                    
+                    
+                //System.out.println("Refitted track chi2 " + gblTrk.getChi2());
+                refittedTracks.add(gblTrk);
+                kinkDataCollection.add(newTrack.getSecond());
+                kinkDataRelations.add(new BaseLCRelation(newTrack.getSecond(), gblTrk));
+            }
+                
+            trajForMPII.delete();
+            trajForMPII_unconstrained.delete();
+                
+
             
         }//loop on tracks
         
         
         if (correctTrack) {
             /*
-            System.out.print("Refitted tracks (" + outputCollectionName + ") N hits: ");
-            for (Track trk : refittedTracks) {
+              System.out.print("Refitted tracks (" + outputCollectionName + ") N hits: ");
+              for (Track trk : refittedTracks) {
               System.out.print(trk.getTrackerHits().size()+" ");
-            }
-            System.out.println();
-             */
+              }
+              System.out.println();
+            */
 
             // Put the tracks back into the event and exit
             int flag = 1 << LCIOConstants.TRBIT_HITS;

--- a/tracking/src/main/java/org/hps/recon/tracking/gbl/SimpleGBLTrajAliDriver.java
+++ b/tracking/src/main/java/org/hps/recon/tracking/gbl/SimpleGBLTrajAliDriver.java
@@ -570,7 +570,7 @@ public class SimpleGBLTrajAliDriver extends Driver {
                 
                 //Align with tracks with at least 6 hits
                 if ((tanLambda > 0 && track.getTrackerHits().size() < actualHitCut) 
-                    || (tanLambda < 0 && track.getTrackerHits().size() < actualHitCut))  {
+                        || (tanLambda < 0 && track.getTrackerHits().size() < actualHitCut))  {
                     continue;
                 }
                 

